src/Model/ItemList.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\ContentApiBundle\Model;
+
+use Countable;
+use IteratorAggregate;
+use Traversable;
+use function count;
+
+final class ItemList implements Countable, IteratorAggregate
+{
+    private $items;
+    private $nextId;
+
+    /**
+     * @param ItemId[] $items
+     */
+    public function __construct(array $items, ?ItemId $nextId)
```

Comment:
This assumption of the data source being an in-memory array points to a more specific name for this class, e.g. `InMemory` or `Array` prefix